esp/platform/txsummary.cpp

```diff
@@ -65,100 +398,443 @@ bool CTxSummary::contains(const char* key) const
+    entry.setown(new TxEntryBool(key, value, logLevel, group));
+    m_entries2.push_back(entry);
+
+    return true;
+}
+
+bool CTxSummary::append(const char* key, int value, const LogLevel logLevel, const unsigned int group, const char* suffix)
+{
+    if(!validateKey(key))
+        return false;
 
     CriticalBlock block(m_sync);
-    Entries::iterator it = find_if(m_entries.begin(), m_entries.end(), MATCH_KEY);
 
-    if (it != m_entries.end())
+    if (__contains(key))
+        return false;
+
+    EntryValue entry;
+    entry.setown(new TxEntryInt(key, value, logLevel, group, suffix));
+    m_entries2.push_back(entry);
+
+    return true;
+}
+
+bool CTxSummary::append(const char* key, __int64 value, const LogLevel logLevel, const unsigned int group, const char* suffix)
+{
+    if(!validateKey(key))
+        return false;
+
+    CriticalBlock block(m_sync);
+
+    if (__contains(key))
+        return false;
+
+    EntryValue entry;
+    entry.setown(new TxEntryInt64(key, value, logLevel, group, suffix));
+    m_entries2.push_back(entry);
+
+
+    return true;
+}
+
+bool CTxSummary::append(const char* key, unsigned int value, const LogLevel logLevel, const unsigned int group, const char* suffix)
+{
+    if(!validateKey(key))
+        return false;
+
+    CriticalBlock block(m_sync);
+
+    if (__contains(key))
+        return false;
+
+    EntryValue entry;
+    entry.setown(new TxEntryUInt(key, value, logLevel, group, suffix));
+    m_entries2.push_back(entry);
+
+    return true;
+}
+
+bool CTxSummary::append(const char* key, unsigned __int64 value, const LogLevel logLevel, const unsigned int group, const char* suffix)
+{
+    if(!validateKey(key))
+        return false;
+
+    CriticalBlock block(m_sync);
+
+    if (__contains(key))
+        return false;
+
+    EntryValue entry;
+    entry.setown(new TxEntryUInt64(key, value, logLevel, group, suffix));
+    m_entries2.push_back(entry);
+
+
+    return true;
+}
+
+bool CTxSummary::append(const char* key, double value, const LogLevel logLevel, const unsigned int group, const char* suffix)
+{
+    if(!validateKey(key))
+        return false;
+
+    CriticalBlock block(m_sync);
+
+    if (__contains(key))
+        return false;
+
+    EntryValue entry;
+    entry.setown(new TxEntryDouble(key, value, logLevel, group, suffix));
+    m_entries2.push_back(entry);
+
+
+    return true;
+}
+
+// Searches for the entry named 'key' and replaces it with the new entry 'value'
+// If not found, then it adds the new entry 'value' to the end of the list
+void CTxSummary::set(const char* key, TxEntryBase* value)
+{
+    bool found = false;
+    EntryValue entry;
+    entry.setown(value);
+    for(auto entryItr = m_entries2.begin(); entryItr != m_entries2.end(); entryItr++)
     {
-        it->value.set(value);
-        it->logLevel = logLevel;
+        if(strieq(key, entryItr->get()->fullname.get()))
+        {
+            found = true;
+            auto finger = m_entries2.erase(entryItr);
```

Comment:
What is the rationale for allowing an atomic entry to replace a named timer, but querying a named timer can't replace an atomic entry?

Reply:
I feel like the name queryTimer implies that you want to operate on a timer exclusively, whereas setEntry() needs to be able to operate on any entry type. Also it is now private.